src/hal/src/device.rs

```diff
@@ -714,6 +714,56 @@ pub trait Device<B: Backend>: fmt::Debug + Any + Send + Sync {
     /// validation layers that can print a friendly name when referring to objects in error messages
     unsafe fn set_pipeline_layout_name(&self, pipeline_layout: &mut B::PipelineLayout, name: &str);
 
+    /// Create external buffer
+    unsafe fn create_external_buffer(
+        &self,
+        external_memory_type_flags: external_memory::ExternalMemoryTypeFlags,
+        usage: buffer::Usage,
+        sparse: memory::SparseFlags,
+        size: u64,
+    ) -> Result<B::Buffer, external_memory::ExternalBufferCreateError>;
+
+    /// Allocate external memory
+    unsafe fn allocate_exportable_memory(
+        &self,
+        external_memory_types: external_memory::ExternalMemoryTypeFlags,
```

Comment:
let's call things consistently, the other place calls this `external_memory_type_flags`

Reply:
Right. Corrected.